tools/src/test/java/org/apache/kafka/tools/consumer/group/share/ShareGroupStateMessageFormatterTest.java

```diff
@@ -164,12 +164,16 @@ private static Stream<Arguments> exceptions() {
             Arguments.of(
                 MessageUtil.toVersionPrefixedByteBuffer((short) 0, SHARE_SNAPSHOT_KEY).array(),
                 MessageUtil.toVersionPrefixedByteBuffer((short) 0, SHARE_UPDATE_VALUE).array(),
-                new RuntimeException("non-nullable field stateBatches was serialized as null")
+                new RuntimeException("Could not read record at offset 0 due to: " +
+                        "Could not read record with version 0 from value's buffer due to: " +
+                        "non-nullable field stateBatches was serialized as null.")
             ),
             Arguments.of(
                 MessageUtil.toVersionPrefixedByteBuffer((short) 1, SHARE_UPDATE_KEY).array(),
                 MessageUtil.toVersionPrefixedByteBuffer((short) 0, SHARE_SNAPSHOT_VALUE).array(),
-                new RuntimeException("non-nullable field stateBatches was serialized as null")
+                new RuntimeException("Could not read record at offset 0 due to: " +
+                        "Could not read record with version 0 from value's buffer due to: " +
+                        "non-nullable field stateBatches was serialized as null.")
             )
         );
     }
```

Comment:
Could we remove `testShareGroupStateMessageFormatter`? We have it in the parent class now.

Reply:
Thank you! I've addressed all the comments!